common/src/main/java/com/redhat/cloud/notifications/models/dto/v1/endpoint/EndpointDTO.java

```diff
@@ -0,0 +1,170 @@
+package com.redhat.cloud.notifications.models.dto.v1.endpoint;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.databind.PropertyNamingStrategies;
+import com.fasterxml.jackson.databind.annotation.JsonNaming;
+import com.redhat.cloud.notifications.models.dto.v1.endpoint.properties.CamelPropertiesDTO;
+import com.redhat.cloud.notifications.models.dto.v1.endpoint.properties.EndpointPropertiesDTO;
+import com.redhat.cloud.notifications.models.dto.v1.endpoint.properties.SystemSubscriptionPropertiesDTO;
+import com.redhat.cloud.notifications.models.dto.v1.endpoint.properties.WebhookPropertiesDTO;
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.AssertTrue;
+import jakarta.validation.constraints.Min;
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Size;
+import org.eclipse.microprofile.openapi.annotations.media.Schema;
+
+import java.time.LocalDateTime;
+import java.util.UUID;
+
+@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
+public final class EndpointDTO {
+    @JsonProperty(access = JsonProperty.Access.READ_ONLY)
+    private UUID id;
+
+    @JsonProperty(required = true)
+    @NotNull
+    @Size(max = 255)
+    private String name;
+
+    @JsonProperty(required = true)
+    @NotNull
```

Comment:
What's the difference for a field between `@NotNull` and `@JsonProperty(required = true) @NotNull`? I only see `@JsonProperty(required = true)` annotations in this class while other DTOs only rely on `@NotNull`. All DTOs should probably be consistent.

Reply:
Yes, the `@NotNull` annotation is enough. I removed the other ones!